src/widgets/histogram/chart.js

```diff
@@ -497,15 +493,22 @@ module.exports = cdb.core.View.extend({
     this.yScale = this._originalYScale;
   },
 
+  _getDataForScales: function () {
+    if (this.model.get('bounded')) {
+      return this.model.get('data');
+    } else {
+      return this._originalData && this._originalData.getData();
+    }
+  },
+
   _setupScales: function () {
+    var data = this._getDataForScales();
```

Comment:
Scale of the histogram depends on the state of itself, if it is bounded (/zoomed) or not.